### ISSUE TITLE
[Feature Request]  Split String

### ISSUE THREAD

**Issue by PP6464** [modularml/mojo#1371]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

There is no method on a string to split it, like there is in literally every programming language. I would like for there to be a function to split a string with a delimiter in mojo

### What is your motivation for this change?

I need it to read data from files and to process input from consoles.

### Any other details?

The method needs to at least be able to split by a delimiter, but can preferably also use regexes

**Comment by JoeLoser** [comment]
Closing as this was implemented and released today in 0.6. Feel free to open a separate issue specifically for regex support if you'd like.

**Comment by cccs-km** [comment]
I am trying to use the `String` `split` method, and it does not seem to be working. I am reading a file in (which is individual words on their own line) and using `.split('/n')` doesn't return the expected output of a DynamicVector[String] for each word. It just returns the entire file for the first element. Am I missing something about the functionality of this method?

```python
let os = Python.import_module("os")

let verbs_file = os.path.join(os.path.expanduser('~'),'workspace/VectorDB/speech/verbs/31K verbs.txt')
with open(verbs_file, 'r') as f:
    verbs = f.read()
let v = verbs.split("/n")
print(v[0])
```
